File: Recursion I/K_symbol_Grammar.cpp
class Solution {
public:
    int kthGrammar(int n, int k) {
        if(n == 1 && k == 1)
            return 0;
        int midterm = pow(2, n - 1) / 2;
        if(k <= midterm)
            return kthGrammar(n - 1, k);
        else 
            return !kthGrammar(n - 1, abs(k - midterm));        
    }
};

File: Recursion I/K_symbol_Grammar.py
class Solution:
    def kthGrammar(self, n: int, k: int) -> int:
        if(n == 1 and k == 1):
            return 0
        midterm = pow(2, n - 1) / 2
        if(k <= midterm):
            return self.kthGrammar(n - 1, k)
        else: 
            if self.kthGrammar(n - 1, abs(k - midterm)) == 0:
                return 1
            else: return 0
        